interface/wx/grid.h

```diff
@@ -1816,6 +1816,167 @@ class wxGridCellCoords
+    /**
+        Set the row of the left top corner.
+     */
+    void SetTopRow(int row);
+
+    /**
+        Return the column of the left top corner.
+     */
+    int GetLeftCol() const;
+
+    /**
+        Set the column of the left top corner.
+     */
+    void SetLeftCol(int col);
+
+    /**
+        Return the row of the bottom right corner.
+     */
+    int GetBottomRow() const;
+
+    /**
+        Set the row of the bottom right corner.
+     */
+    void SetBottomRow(int row);
+
+    /**
+        Return the column of the bottom right corner.
+     */
+    int GetRightCol() const;
+
+    /**
+        Set the column of the bottom right corner.
+     */
+    void SetRightCol(int col);
+
+    /**
+        Set the row and column of the coordinate.
+     */
+    void Set(int topRow, int leftCol, int bottomRow, int rightCol);
+
+    /**
+        Return the coordinates of the top left corner.
+     */
+    wxGridCellCoords GetTopLeft() const
+    {
+        return wxGridCellCoords(m_topRow, m_leftCol);
+    }
+
+    /**
+        Return the coordinates of the bottom right corner.
+     */
+    wxGridCellCoords GetBottomRight() const
+    {
+        return wxGridCellCoords(m_bottomRow, m_rightCol);
+    }
+
+    /**
+        Equality operator.
+     */
+    bool operator==(const wxGridBlockCoords& other) const;
+
+    /**
+        Inequality operator.
+     */
+    bool operator!=(const wxGridBlockCoords& other) const;
+
+    /**
+        Checks whether the cells block is invalid.
+
+        Returns false only if all components are -1. Notice that if one
+        of components (but not all) is -1, this method returns true even if
+        the object is invalid. This is done because objects in such state
+        should actually never exist, i.e. either all components should be -1
+        or none of them should be -1.
+     */
+    bool operator!() const;
+
+private:
+    int m_topRow;
+    int m_leftCol;
+    int m_bottomRow;
+    int m_rightCol;
+};
+
+/**
+    The iterator for grid selection blocks.
+ */
+class wxGridSelectionIterator
+{
+public:
+    /**
+        Default constructor initializes the iterator.
+     */
+    wxGridSelectionIterator();
+
+    /**
+     * Get the current selection block coordinates.
+     */
+    const wxGridBlockCoords& GetBlockCoords() const;
+
+    /**
+     * Iterate to the next block.
+     */
+    void Next();
+
+    /**
+     * Whether the iterator is valid.
+     */
+    bool Valid();
+
+    /**
+     * The function to allow using range-based for.
+     */
+    wxGridBlockCoords *begin() const;
+
+    /**
+     * The function to allow using range-based for.
+     */
+    wxGridBlockCoords *end() const;
+
+private:
+    /**
+        Private constructor used by wxGrid to construct the iterator.
+     */
+    wxGridSelectionIterator(wxGridBlockCoords *begin, wxGridBlockCoords *end);
+
+    wxGridBlockCoords *m_begin;
+    wxGridBlockCoords *m_end;
+    wxGridBlockCoords *m_it;
+
+    friend class wxGrid;
```

Comment:
Private parts don't need to/shouldn't be present in the interface files unless it's particularly useful for some reason (e.g. above it could be considered to be useful because it shows that `wxGridBlockCoords` is just a trivial POD type).
```suggestion
```